### PR TITLE
Fix for MaskedBehavior prevents digit input

### LINKED ISSUE #1276: [Bug] MaskedBehavior prevents input digit
### Description

I'm using the following Mask and UnMaskedCharacter to handle a phone number in an entry field:

Mask="+1 (AAA) AAA-AAAA" UnMaskedCharacter="A"

This Mask prevents the input of the digit "1". I'm not sure if this is the expected behavior, but I was expected to be able to input any digit.

### Steps to Reproduce

```
<Entry x:Name="PhoneEntry" Text="{Binding Phone}" Placeholder="Phone" Keyboard="Telephone">
    <Entry.Behaviors>
        <xct:CharactersValidationBehavior x:Name="PhoneValidator" CharacterType="Digit" MinimumCharacterCount="11" Flags="ValidateOnValueChanging" InvalidStyle="{StaticResource InvalidEntryStyle}" />
        <xct:MaskedBehavior Mask="+1 (AAA) AAA-AAAA" UnMaskedCharacter="A" />
     </Entry.Behaviors>
</Entry>
```

### Expected Behavior

Able to input "1" digit

### Actual Behavior

Unable to input "1" digit

### Basic Information

- Version with issue: Xamarin Toolkit 1.1
- Last known good version:
- IDE: VS 2019 Mac
- Platform Target Frameworks: 
  - iOS:  14.5
  - Android: 9, 10, 11

### PR DESCRIPTION
**Pull request by IeuanWalker**
### Description of Change ###

Fixes `RemoveMask` method. 
Originally it wasn't taking into account the position of the user's input and removing it if it was part of the mask.

### Bugs Fixed ###

- Fixes #1276 

### API Changes ###

It now compares the char position to the mask position.

### Behavioral Changes ###

None that I can think of, all other samples work as before.

### PR Checklist ###
- [x] Has a linked Issue, and the Issue has been `approved`
- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: samples/XCT.Sample/Pages/Behaviors/MaskedBehaviorPage.xaml [hunk 1]
```diff
@@ -71,6 +71,19 @@
...
+                   Placeholder="Phone">
+                <Entry.Behaviors>
+                    <xct:CharactersValidationBehavior x:Name="PhoneValidator"
+                                                      CharacterType="Digit"
+                                                      Flags="ValidateOnValueChanging"
+                                                      MinimumCharacterCount="11" />
+                    <xct:MaskedBehavior Mask="+1 (AAA) AAA-AAAA" UnMaskedCharacter="A" />
+                </Entry.Behaviors>
+            </Entry>
+            <Label Padding="10,0,10,50" FontSize="Small" Text="Test for issue #1276'" /> <!-- https://github.com/xamarin/XamarinCommunityToolkit/issues/1276 -->
```
Comment (AndreiMisiukevich) [inline samples/XCT.Sample/Pages/Behaviors/MaskedBehaviorPage.xaml:86]: We have "Test Cases" folder for this purpose :)

**Comment by IeuanWalker** [comment]
> @IeuanWalker thanks for the quick fix! Your code doesn't allow me to type `1` as the first number. Can you review it?
> 
> Also, will be great if you add unit tests for it

@pictos you can now enter 1, but it does count as the masked character, i.e. text would become `+1`
where as if you typed 8 for example it wouldn't, i.e. text value would become `+1 (8`

Atm not sure how to get around that.
Let me know what you think.

Will look at adding some tests.

File: samples/XCT.Sample/Pages/Behaviors/MaskedBehaviorPage.xaml (hunk 1 above)
Comment (IeuanWalker) [reply samples/XCT.Sample/Pages/Behaviors/MaskedBehaviorPage.xaml:86]: @AndreiMisiukevich lol never noticed it, will look at moving it

**Comment by jfversluis** [comment]
Ping us when you think it's done for another review @IeuanWalker

**Comment by IeuanWalker** [comment]
@jfversluis @pictos @AndreiMisiukevich 

Think this is ready for review.

One thing to note, the unit test behaves differently than the sample app _(it's what I mentioned in my previous comment)_

This is the test -
```csharp
[Test]
public void NumbersInMaskShouldBeAllowed()
{
	// arrange
	var entry = CreateEntry("+1 (AAA) AAA-AAAA", 'A');

	// act
	entry.Text = "12345";

	// assert
	Assert.IsTrue(entry.Text.Equals("+1 (123) 45"));
}
```

But when running the sample app this is the outcome - `+1 (234) 5`.
I can't see why this is happening, but this PR does fix the issue raised + doesn't change any existing behaviour, so I think it's ok.

**Comment by pictos** [comment]
@IeuanWalker I added a unitTest that reproduces the error, maybe that helps. I'll try to find some solution for this, but if we can't the best is to say that we don't have support for such a feature.

**Comment by jfversluis** [comment]
Seems like the build is failing atm

**Comment by jsuarezruiz** [comment]
Could you rebase to fix the merge conflicts?

**Review by AndreiMisiukevich — CHANGES_REQUESTED** [review]
Rebase needed

**Comment by IeuanWalker** [comment]
@AndreiMisiukevich sorry not quite sure how to rebase

**Comment by jfversluis** [comment]
Now that we're so close to the sunsetting of Xamarin (and therefore also Xamarin Community Toolkit) unfortunately we won't be able to take this in anymore, we're really sorry about that. Nevertheless, thank you so much for your time and effort that you have put into this PR, we really appreciate it.

Please have a look at the evolution of the Xamarin Community Toolkit, [The .NET MAUI Community Toolkit](https://github.com/CommunityToolkit/Maui). A lot of development has been going on there. Hopefully this issue was already fixed in that codebase. If not, feel free to port this over to there.

Again, thank you so much for being a contributor and Xamarin user!

**Pull request closed without merging**